### PR TITLE
471 remove `CodeClass` from `FilteredData`

### PR DESCRIPTION
**Pull request by chlebowa**
Preprocessing code is no longer kept in `FilteredData$code` but attached to a `FilteredData` as an attribute.
Necessary adjustments were made in `teal_data_to_filtered_data` and in `get_datasets_code`.

[companion](https://github.com/insightsengineering/teal.slice/pull/488)

### CODE REVIEW

**Review by gogonzo — APPROVED** [review]

File: R/module_nested_tabs.R [hunk 1]
```diff
@@ -311,8 +311,7 @@ srv_nested_tabs.teal_module <- function(id, datasets, modules, is_module_specifi
   )
 
   hashes <- calculate_hashes(datanames, datasets)
-  metadata <- lapply(datanames, datasets$get_metadata)
-  names(metadata) <- datanames
+  metadata <- sapply(datanames, datasets$get_metadata, simplify = FALSE)
```
[the remark this answers is not in the document]
Comment (chlebowa) [reply R/module_nested_tabs.R:314]: It is a public method, though :thinking: 

There is a `get_metadata` function in `teal` with `tdata` and `default` methods. Perhaps we should use that?

[the remark this answers is not in the document]
Comment (chlebowa) [reply R/module_nested_tabs.R:314]: Oh, one more thing. On main this function looks like this:

```
.datasets_to_data <- function(module, datasets) {
  checkmate::assert_class(module, "teal_module")
  checkmate::assert_class(datasets, "FilteredData")

  datanames <- if (is.null(module$datanames) || identical(module$datanames, "all")) {
    datasets$datanames()
  } else {
    unique(module$datanames) # todo: include parents! unique shouldn't be needed here!
  }

  # list of reactive filtered data
  data <- sapply(datanames, function(x) datasets$get_data(x, filtered = TRUE), simplify = FALSE)

  hashes <- calculate_hashes(datanames, datasets)

  code <- c(
    get_rcode_str_install(),
    get_rcode_libraries(),
    get_datasets_code(datanames, datasets, hashes),
    teal.slice::get_filter_expr(datasets, datanames)
  )

  do.call(
    teal.data::teal_data,
    args = c(data, code = list(code), join_keys = list(datasets$get_join_keys()[datanames]))
  )
}
```
No metadata handling at all.

[the remark this answers is not in the document]
Comment (chlebowa) [reply R/module_nested_tabs.R:314]: Maybe this just is a merge artefact?

[the remark this answers is not in the document]
Comment (gogonzo) [reply R/module_nested_tabs.R:314]: > No metadata handling at all.

You copied the function not from `main`. On main `.datasets_to_data` looks like this:

https://github.com/insightsengineering/teal/blob/50bbab6d74a1a0e80057a42f262846156962f584/R/module_nested_tabs.R#L314

[the remark this answers is not in the document]
Comment (chlebowa) [reply R/module_nested_tabs.R:314]: My mistake. The function above is on `refactor` branch.

**Pull request merged**